iphone/Maps/UI/PlacePage/PlacePageInteractor.swift

```diff
@@ -242,6 +238,13 @@ extension PlacePageInteractor: PlacePageHeaderViewControllerDelegate {
   func previewDidPressExpand() {
     presenter?.showNextStop()
   }
+
+  func previewDidPressShare(from sourceView: UIView) {
+    if let shareVC = ActivityViewController.share(forPlacePage: placePageData),
```

Comment:
Why is it under if? Can it be nil? If yes, then what is this case?

Reply:
for some reason (i dunno why) all the initializers for the activity VC are marked as a `nullable` directly:
```obj
NS_SWIFT_NAME(ActivityViewController)
@interface MWMActivityViewController : UIActivityViewController

+ (nullable instancetype)shareControllerForEditorViral;

+ (nullable instancetype)shareControllerForMyPosition:(CLLocationCoordinate2D)location;

+ (nullable instancetype)shareControllerForPlacePage:(PlacePageData *)data;

+ (nullable instancetype)shareControllerForURL:(nullable NSURL *)url
                              message:(NSString *)message
                    completionHandler:(nullable UIActivityViewControllerCompletionWithItemsHandler)completionHandler;

- (void)presentInParentViewController:(UIViewController *)parentVC anchorView:(nullable UIView *)anchorView;

@end
```
the initializing process involves c++ data and some asserts for the object's absence:
```

- (instancetype)initForPlacePage:(PlacePageData *)data {
  self = [super init];
  if (self)
  {
    NSAssert(data, @"Entity can't be nil!");
    _isMyPosition = data.isMyPosition;
    _data = data;
  }
  return self;
}
```
It seems like the idea was to not crash the app in prod

Reply:
> It seems like the idea was to not crash the app in prod.

It sounds strange. If the property can't be null, then nullable should be removed. It's better to crash immediately and find a fix instead of getting user reports that something doesn't work.